apollo-mockserver/src/commonMain/kotlin/com/apollographql/apollo3/mockserver/MockServerCommon.kt

```diff
@@ -89,13 +125,15 @@ internal fun readRequest(source: BufferedSource): MockRequest? {
 
   val contentLength = headers["Content-Length"]?.toLongOrNull() ?: 0
   val transferEncoding = headers["Transfer-Encoding"]?.lowercase()
-  check(transferEncoding == null || transferEncoding == "identity") {
+  check(transferEncoding == null || transferEncoding == "identity" || transferEncoding == "chunked") {
     "Transfer-Encoding $transferEncoding is not supported"
   }
 
   val buffer = Buffer()
   if (contentLength > 0) {
     source.read(buffer, contentLength)
+  } else if (transferEncoding == "chunked") {
+    source.readChunked(buffer)
```

Comment:
So what happens inside OkHttp is that if the request's body's size is `-1` then it automatically switches to chunked. This is the case when using batching, and also with Uploads.

So now we need to allow chunked on the JVM, otherwise the related tests fail 😅. 

I guess this behavior doesn't happen with `NSMutableURLRequest` which is why this didn't cause any issues on Apple.

Also (sorry this is a bit off-topic!) but while looking at [this](https://github.com/apollographql/apollo-kotlin/blob/3769fe5edd7010f14cde03bb57235628c8d13328/apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/network/http/BatchingHttpInterceptor.kt#L136), I noticed I have the impression that the size will always be -1 (this header is not normally present, right?)

Reply:
Ahhh, indeed, I see Upload size might not all be known in advance indeed 👍 

For the other queries. I think we should have `Content-Length` always? This is happening [here](https://github.com/apollographql/apollo-kotlin/blob/4b02a22ff2b39b9c2aa05a68aaff467276fc8669/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/http/DefaultHttpRequestComposer.kt#L209)

Reply:
Well this sets the `contentLength` on the `HttpBody` but not a `Content-Length` header. (Unless I'm mistaken!) that's added later not by our code but by OkHttp.

So I think the line in `BatchingHttpInterceptor` should be replaced by:
```kotlin
val allLengths = pending.map { it.request.body?.contentLength ?: 0}
```